Wikipedia/Code/UITextView+Extensions.swift

```diff
@@ -15,4 +15,55 @@ extension UITextView {
         
         return rect
     }
+
+    /// Adds formatting characters around selected text or around cursor
+    /// - Parameters:
+    ///   - formattingString: string used for formatting, will surround selected text or cursor
+    ///   - cursorOffset: offset to adjust cursor position, should be equal to the number of characters
+    func addStringFormattingCharacters(formattingString: String) {
+        let cursorOffset = formattingString.count
+        if let selectedRange = selectedTextRange {
+            let cursorPosition = offset(from: endOfDocument, to: selectedRange.end)
+            if selectedRange.isEmpty {
+                replace(selectedTextRange ?? UITextRange(), withText: formattingString + formattingString)
```

Comment:
I can't quite place what happened but I did prefer the previous way toggling on and off formatting worked when it was _only_ a cursor. Now it jumps to the end of the formatting ticks, but before it stayed in the middle for easy toggling.

Toggling when text is highlighted is working great for me now though! Hopefully you can restore the single cursor behavior without affecting that.